imap/mailbox.h

```diff
@@ -339,36 +343,39 @@ struct mailbox_iter;
-#define OFFSET_DELETEDMODSEQ 76    /* CONDSTORE (64-bit modseq) */
-#define OFFSET_EXISTS 84           /* Non-expunged records */
+#define OFFSET_EXISTS 60           /* Non-expunged records */
+#define OFFSET_MAILBOX_OPTIONS 64
+#define OFFSET_LEAKED_CACHE 68     /* Number of leaked records in cache file */
+#define OFFSET_HIGHESTMODSEQ 72    /* CONDSTORE (64-bit modseq) */
+#define OFFSET_DELETEDMODSEQ 80    /* CONDSTORE (64-bit modseq) */
 #define OFFSET_FIRST_EXPUNGED 88   /* last_updated of oldest expunged message */
 #define OFFSET_LAST_REPACK_TIME 92 /* time of last expunged cleanup  */
 #define OFFSET_HEADER_FILE_CRC 96  /* CRC32 of the index header file */
 #define OFFSET_SYNCCRCS_BASIC 100  /* XOR of SYNC CRCs of unexpunged records */
 #define OFFSET_RECENTUID 104       /* last UID the owner was told about */
 #define OFFSET_RECENTTIME 108      /* last timestamp for seen data */
-#define OFFSET_POP3_SHOW_AFTER 112 /* time after which to show messages
+#define OFFSET_POP3_SHOW_AFTER 112 /* time after which to show messages 
                                     * to POP3 */
 #define OFFSET_QUOTA_ANNOT_USED 116 /* bytes of per-mailbox and per-message
                                      * annotations for this mailbox */
-                          /* Spares - only use these if the index */
-                          /*  record size remains the same */
-#define OFFSET_SYNCCRCS_ANNOT 120 /* SYNC_CRC of the annotations */
-#define OFFSET_UNSEEN 124         /* total number of UNSEEN messages (owner) */
-/* NEXT UPDATE - add Bug #3562 "TOTAL_MAILBOX_USED" field, 64 bit
- * value which counts the total size of all files included expunged
- * files. We've created the header space now, but will also need code
- * changes, so holding off */
+#define OFFSET_SYNCCRCS_ANNOT 120  /* SYNC_CRC of the annotations */
+#define OFFSET_UNSEEN 124          /* total number of UNSEEN messages (owner) */
 #define OFFSET_MAILBOX_CREATEDMODSEQ 128 /* MODSEQ at creation time */
-#define OFFSET_CHANGES_EPOCH 136   /* time from which we can calculate changes */
-#define OFFSET_SPARE1 140
-#define OFFSET_SPARE2 144
-#define OFFSET_SPARE3 148
-#define OFFSET_SPARE4 152
-#define OFFSET_HEADER_CRC 156 /* includes all zero for the spares! */
+#define OFFSET_QUOTA_DELETED_USED 136  /* bytes of \Deleted messages
+                                        * for this mailbox (64-bit) */
+#define OFFSET_QUOTA_EXPUNGED_USED 144 /* bytes of \Expunged messages
+                                        * for this mailbox (64-bit) */
+#define OFFSET_CHANGES_EPOCH 152   /* time from which we can calculate changes */
+#define OFFSET_HEADER_CRC 156
+
+#define ORIG_OFFSET_MAILBOX_OPTIONS 60
+#define ORIG_OFFSET_LEAKED_CACHE 64
+#define ORIG_OFFSET_HIGHESTMODSEQ 68
+#define ORIG_OFFSET_DELETEDMODSEQ 76
+#define ORIG_OFFSET_EXISTS 84
+#define ORIG_OFFSET_CHANGES_EPOCH 136
+#define ORIG_OFFSET_QUOTA_DELETED_USED 140
+#define ORIG_OFFSET_QUOTA_EXPUNGED_USED 148
```

Comment:
If we called these like `PRE19_OFFSET_...` instead of `ORIG_OFFSET_...`, then that might make it easier to manage next time we need to rearrange them, and a little less ambiguous.

I think a similar thing would work for the `struct index_header_offsets`s too, and then I think the code that switches between them would look a bit nicer then too, like: `if (i.minor_version < 19) offsets = &pre19_offsets;`, or however it went.

Alternatively, we could not worry about it now, and figure it out when we need to.

Reply:
Yep, I had the same idea as I was reading your other comment.  Will rename